### DISCUSSION TITLE
How does one limit the amount of items GetVideosAsync returns?

### DISCUSSION THREAD

**Discussion by Kajirinn** [Tyrrrz/YoutubeExplode#559] [Q&A] [answered]
Hi there,

I've used this library for a while now, but after taking a short break I've noticed there are changes within the search component. Before these changes it used to work with pages, which made searches limited but relatively quick.

I'm not really sure how the new search works. I understand that by awaiting the foreach, iterating occurs in parallel, but that also makes it fairly difficult to limit the amount of results I want. Average search time has increased by literal minutes now, since I get roughly 600 results back.

Any idea on how to limit my results to roughly 1 page with an IAsyncEnumerable, would be greatly appreciated.

Thank you advance,
Kajirinn.

**Comment by Kajirinn** [comment, marked as answer]
Never mind, I've found the answer after looking into the issue for a while longer. Somehow I've completely missed the existence of 'CollectAsync'.

**Reply by Tyrrrz** [reply]
Also, see `GetVideoBatchesAsync(...)` as it returns videos batched by pages, meaning each iteration is exactly one page.
https://github.com/Tyrrrz/YoutubeExplode#getting-videos-included-in-a-playlist

**Reply by Kajirinn** [reply]
Thank you for the additional information, I'll definitely take a look! Always appreciate your insanely quick responses, thank you for keeping this library up-to-date.
